lighthouse-core/runner.js

```diff
@@ -150,38 +137,68 @@ class Runner {
    * @param {(runnerData: {config: TConfig, driverMock?: Driver}) => Promise<LH.Artifacts>} gatherFn
-   * @param {{config: TConfig, driverMock?: Driver}} options
+   * @param {{config: TConfig, driverMock?: Driver, computedCache: Map<string, ArbitraryEqualityMap>}} options
    * @return {Promise<LH.Artifacts>}
    */
-  static async gatherAndManageArtifacts(gatherFn, options) {
+  static async gather(gatherFn, options) {
     const settings = options.config.settings;
 
     // Gather phase
     // Either load saved artifacts from disk or from the browser.
-    let artifacts;
-    if (settings.auditMode && !settings.gatherMode) {
-      // No browser required, just load the artifacts from disk.
-      const path = this._getDataSavePath(settings);
-      artifacts = assetSaver.loadArtifacts(path);
-      const requestedUrl = artifacts.URL.requestedUrl;
-
-      if (!requestedUrl) {
-        throw new Error('Cannot run audit mode on empty URL');
-      }
-    } else {
-      artifacts = await gatherFn({
-        config: options.config,
-        driverMock: options.driverMock,
+    try {
+      const runnerStatus = {msg: 'Gather phase', id: 'lh:runner:gather'};
+      log.time(runnerStatus, 'verbose');
+
+      const sentryContext = Sentry.getContext();
+      Sentry.captureBreadcrumb({
+        message: 'Run started',
+        category: 'lifecycle',
+        data: sentryContext?.extra,
       });
 
-      // -G means save these to disk (e.g. ./latest-run).
-      if (settings.gatherMode) {
+      /** @type {LH.Artifacts} */
+      let artifacts;
+      if (settings.auditMode && !settings.gatherMode) {
+        // No browser required, just load the artifacts from disk.
         const path = this._getDataSavePath(settings);
-        await assetSaver.saveArtifacts(artifacts, path);
+        artifacts = assetSaver.loadArtifacts(path);
+        const requestedUrl = artifacts.URL.requestedUrl;
+
+        if (!requestedUrl) {
+          throw new Error('Cannot run audit mode on empty URL');
+        }
+      } else {
+        artifacts = await gatherFn({
+          config: options.config,
+          driverMock: options.driverMock,
+        });
+
+        // -G means save these to disk (e.g. ./latest-run).
+        if (settings.gatherMode) {
+          const path = this._getDataSavePath(settings);
+          await assetSaver.saveArtifacts(artifacts, path);
+        }
       }
+
+      log.timeEnd(runnerStatus);
+
+      return artifacts;
+    } catch (err) {
+      throw await Runner.createRunnerError(err, settings);
     }
+  }
 
-    return artifacts;
+  /**
+   * @param {any} err
+   * @param {LH.Config.Settings} settings
+   */
+  static async createRunnerError(err, settings) {
+    // i18n LighthouseError strings.
+    if (err.friendlyMessage) {
+      err.friendlyMessage = format.getFormatted(err.friendlyMessage, settings.locale);
+    }
+    await Sentry.captureException(err, {level: 'fatal'});
```

Comment:
this function could be sync, and we can do `Sentry.captrueException(...).catch(() => console.log('failed to send error to sentry'))` (or just do nothing).

We really don't want to have a fatal error just because `Sentry.captureException` decided to throw.